### PR TITLE
feat(QIntersection): new prop "root"

### LINKED ISSUE #7851: q-intersection examples seem to be not working
Hi There,
Just a quick update that the examples behave odd for q-intersection:
https://quasar.dev/vue-components/intersection#QIntersection-API

When visit Codepen from the links provided in Docs the displays are empty (there is a split second content load on the initial page render, though).

For recreating, just try to follow to Codepen from the first two examples in docs to see that empty screen.

Checked on Google Chrome, Firefox and Safari.

Cheers!

### PR DESCRIPTION
**Pull request by hawkeye64**
Uses Intersection directive "root" config. Using the "root" property allows an ancestor to be defined. Currently, none of the QIntersection/Intersection examples work on Codepen. I have added one for QIntersection that will work because it uses it's immediate parent as root.

### CODE REVIEW

**Comment by rstoenescu** [comment]
Kick-ass work!

File: ui/src/components/intersection/QIntersection.js [hunk 1]
```diff
@@ -26,6 +26,10 @@ export default Vue.extend({
 
     margin: String,
     threshold: [ Number, Array ],
+    root: {
+      type: Element,
```
Comment (rstoenescu) [inline ui/src/components/intersection/QIntersection.js:30]: Did you verify SSR for this?

File: ui/src/components/intersection/QIntersection.json [hunk 2]
```diff
@@ -30,6 +31,14 @@
       "addedIn": "v1.9.5"
     },
 
+    "root": {
+      "type": "Element",
+      "desc": "[Intersection API root prop] Lets you define an alternative to the viewport as your root (through its DOM element); It is important to keep in mind that root needs to be an ancestor of the observed element",
+      "examples": [ "$refs.myTable.$el", "getElementById(\"myTable\")" ],
+      "category": "behavior",
+      "addedIn": "v1.14.1"
```
Comment (rstoenescu) [inline ui/src/components/intersection/QIntersection.json:39]: Should be v1.14.2 now.

**Comment by rstoenescu** [comment]
Left two comments (above).
Can you also port this to the vue3-work branch, pls? @hawkeye64

**Comment by rstoenescu** [comment]
No need to port it. Merge was successful. But we need another PR for dev (and then I'll also merge it to vue3-work). @hawkeye64

**Comment by rstoenescu** [comment]
Please check my two comments on code from above. Remove type: Element (it breaks SSR) and set addedIn to 1.14.2 (v1.14.1 was already out)

**Comment by hawkeye64** [comment]
Other than `Element`, what type should it be?

**Comment by rstoenescu** [comment]
None. Just remove that line. There's no solution working on SSR and non-SSR.

**Pull request merged**